### ISSUE TITLE
EI: In S14 Drowned Plains Replays, Hahid Dialogue Breaks Replays

### ISSUE THREAD

**Issue by IskarJarak01** [wesnoth/wesnoth#8856]
### Game and System Information

- **Version:**1.18
- **OS:**ChromeOS/ Debian

### Description of the bug

An optional ally from one of Eatern Invasion's branching paths (Hahid / Dune Apothecary) gives the player a dialogue choice in S14 The Drowned Plains. 

In replays, it creates multiple out-of-sync errors. The player is then given the dialogue choice as though they were playing and the game crashes when an option is selected.

Out-of-sync errors:
![Screenshot 2024-05-07 13 02 02](https://github.com/wesnoth/wesnoth/assets/81496259/56e6fc9e-e0ed-4928-b765-62f4e0c67725)
![Screenshot 2024-05-07 13 02 41](https://github.com/wesnoth/wesnoth/assets/81496259/e863829d-f5a7-4164-a7b2-5116117d028f)

### Steps to reproduce the behavior

1. Bring Hahid to Drowned Plains
2. Allow Hahid's dialogue to trigger
3. Do something (move, attack etc.) after Hahid's dialogue triggers
4. Save a replay then load it

[EI-The Drowned Plains replay 20240409-134014.gz](https://github.com/wesnoth/wesnoth/files/15236450/EI-The.Drowned.Plains.replay.20240409-134014.gz)

### Additional context

There are other similar dialogue choices from Hahid which work correctly in replays, this issue is probably related to the different trigger for this dialogue.

Dialogue options work correctly when playing, the bug is specific to replays.

**Comment by gfgtdf** [comment]
https://github.com/wesnoth/wesnoth/blob/f657a3654908eea264004f124e7970221bc7b265/data/campaigns/Eastern_Invasion/scenarios/14_The_Drowned_Plains.cfg#L1297

uses a select event.

**Comment by gfgtdf** [comment]
https://github.com/wesnoth/wesnoth/issues/8467 is similar

**Comment by Wedge009** [comment]
Is this and #8467 a situation where #1834 would help?

**Comment by gfgtdf** [comment]
No, that issue is aout displaing additional stuff in replay not about fixing oos errory.

I think the only thing that helps here is to fix the wml code so that it doesn't use select events for this things. Select events are marked as replay-unsafe in the wiki too

**Comment by gfgtdf** [comment]
one could also try to use the `wesnoth.sync.invoke_command` lua function here, but i'm not sure whetehr that'll work
